application.go

```diff
@@ -50,6 +50,7 @@ type Application struct {
 	Mem                   *float64            `json:"mem,omitempty"`
 	Tasks                 []*Task             `json:"tasks,omitempty"`
 	Ports                 []int               `json:"ports"`
+	PortDefinitions       []*PortDefinition   `json:"portDefinitions,omitempty"`
```

Comment:
I think this needs to be a `*[]PortDefinition` if we want to be able to submit an empty array (in order to, say, clear an existing array of non-empty port definitions). If you agree, then we should also have a (tested) `AddPortDefinition()` helper method (similar to [AddHealthCheck()](https://github.com/gambol99/go-marathon/blob/5e63f97a03e7c519e489834142de7893649aae53/application.go#L286)) to ease usage of pointers.

(See my other comment regarding why we need pointers in the first place.)